verifier/src/lib.rs

```diff
@@ -2,11 +2,260 @@
 
 #![doc = include_str!("../README.md")]
 #![deny(missing_docs, missing_debug_implementations, unsafe_code)]
+#![no_std]
+
+use core::fmt::Debug;
+use subtle::CtOption;
+
+/// Failed to verify.
+#[derive(Debug, Eq, PartialEq)]
+pub struct VerificationError;
+
+/// A verifier. These can chained together using the [`Or`] and [`And`]
+/// types.
+pub trait Verifier: Debug {
+    /// The error that this verification will return in failure cases.
+    type Error;
+
+    /// Performs a verification operation for the [`Verifier`].
+    ///
+    /// In order to accommodate constant time operations this returns a
+    /// [`CtOption`] instead of a [`Result`].
+    fn verify(&self) -> CtOption<Self::Error>;
+}
+
+/// An error that occurs during an `and` operation.
+#[derive(Debug)]
+pub struct AndError<L, R> {
+    _left: CtOption<L>,
+    _right: CtOption<R>,
+}
+
+impl<L, R> AndError<L, R> {
+    /// Create a new instance
+    pub fn new(left: CtOption<L>, right: CtOption<R>) -> Self {
+        Self {
+            _left: left,
+            _right: right,
```

Comment:
Is this prefixed because "written but never read"?

Reply:
That is correct, clippy barked about it.
I think once we implement the display, this should go away